plugins/module_utils/core.py

```diff
@@ -227,9 +227,12 @@ def get_resource(
 
         if isinstance(primary_identifier, list):
             for id in primary_identifier:
-                identifier[
-                    snake_to_camel(id, capitalize_first=True)
-                ] = self.module.params.get(id)
+                if id == "ACLName":
+                    identifier[id] = self.module.params.get("acl_name")
+                else:
+                    identifier[
+                        snake_to_camel(id, capitalize_first=True)
+                    ] = self.module.params.get(id)
```

Comment:
The same part repeated several times, can you move this into a dedicated function, please?